.github/workflows/ci_cd.yml

```diff
@@ -0,0 +1,110 @@
+name: Client & CMS CI/CD
+
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  deploy:
+    name: Deploy Client app
+    # if: (github.head_ref || github.ref_name) == 'main' || (github.head_ref || github.ref_name) == 'staging'
+    # if: (github.head_ref || github.ref_name) == 'staging'
+    if: (github.head_ref || github.ref_name) == 'infrastructure'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Extract branch name
+        shell: bash
+        run: |
+          echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+          echo "##[set-output name=branch-upper;]$(echo ${GITHUB_REF#refs/heads/} | tr a-z A-Z )"
+        id: extract_branch
+
+      - name: Filter changes
```

Comment:
what's the advantage of using this vs the native thing https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Reply:
I need to run workflow, but sometimes I don't need to build images when there are no changes. Filters which you are mentioning can be defined only per workflow right? I think it is not possible to decide which steps can/cannot be run.  I can be wrong, in that case let me know ;).

Reply:
ah, I see the code that does that. Does this step also stop/prevent the deployment in case neither filters pass? for example, if I make a PR with a change just to the README?

Reply:
This version still runs deploy --> restarts beanstalk, but It should be quite straightforward to add all filters to it and skip even this step in case there is no change in any of the docker images or `afoco_source_bundle` that resides inside infrastructure folder...